spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/SQLiteDialect.java

```diff
@@ -0,0 +1,100 @@
+package org.springframework.data.relational.core.dialect;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.data.convert.WritingConverter;
+import org.springframework.data.relational.core.sql.LockOptions;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * SQL Dialect for SQLite.
+ *
+ * @author Rudolf Schmidt
+ * @since 3.3.0
+ */
+public class SQLiteDialect extends AbstractDialect {
+
+	public static final SQLiteDialect INSTANCE = new SQLiteDialect();
+
+	private static final LimitClause LIMIT_CLAUSE = new LimitClause() {
+		@Override
+		public String getLimit(long limit) {
+			return String.format("limit %d", limit);
+		}
+
+		@Override
+		public String getOffset(long offset) {
+			throw new UnsupportedOperationException("offset alone not supported");
```

Comment:
I think we can do better, like `LIMIT -1 OFFSET %d`. [According to official documentation](https://sqlite.org/lang_select.html#the_limit_clause) it is fine to set `LIMIT` to the negative value.